libs/datalayout/src/main/java/com/linkedin/openhouse/datalayout/ranker/GreedyMaxBudgetCandidateSelector.java

```diff
@@ -0,0 +1,39 @@
+package com.linkedin.openhouse.datalayout.ranker;
+
+import com.linkedin.openhouse.datalayout.strategy.DataLayoutStrategy;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.PriorityQueue;
+import org.apache.commons.lang3.tuple.Pair;
+
+/**
+ * A greedy candidate selector that selects the top K data layout strategies based on the max
+ * budget. The max budget is defined by the max estimated compute cost or the max number of tables
+ * whatever reaches first.
+ */
+public class GreedyMaxBudgetCandidateSelector extends BaseDataLayoutCandidateSelector {
+
+  private final double maxEstimatedComputeCost;
+  private final int maxTables;
```

Comment:
Why do we need to set a number of maximum tables as well? Is that a case that you've encountered, that you didn't want to compact too many tables?

Reply:
Just realized when Stas said maxTables case is same as topK when compute cost budget is infinity, so I removed the topK class.

Use-case is it will allow us to ramp DLO at production along two dimensions number of tables, and compute cost (as we understand it better).

I guess eventually, it should be only bound by compute cost.

Reply:
Good to have some safety net

Reply:
Yes, that too.